From 9536e825a52af58e14c42126eb5eee14e398075d Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Tue, 23 Apr 2024 17:41:44 +0200
Subject: [PATCH 0816/1695] plat-stm32mp2: differentiate internal tamper
 activation per SoC

STM32MP21 SoCs do not embed the same internal tampers as the
STM32MP23/25 ones. Differentiate the ones that need specific operations.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 8eade98d5b6aaa62914b8509ed69b39e62c5e5c6 plat-stm32mp2: differentiate internal tamper activation per SoC]
Change-Id: Ifb7db0db3db31c3f8197657700cdb7c13909d3d7
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/376036
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 core/arch/arm/plat-stm32mp2/main.c | 37 ++++++++++++++++++++++++------
 1 file changed, 30 insertions(+), 7 deletions(-)

diff --git a/core/arch/arm/plat-stm32mp2/main.c b/core/arch/arm/plat-stm32mp2/main.c
index 90b840ec7..183d663a4 100644
--- a/core/arch/arm/plat-stm32mp2/main.c
+++ b/core/arch/arm/plat-stm32mp2/main.c
@@ -330,14 +330,38 @@ TEE_Result stm32_activate_internal_tamper(int id)
 			res = TEE_SUCCESS;
 		break;
 
-	case INT_TAMP7: /* VDDCORE monitoring under/over voltage */
-		stm32mp_pwr_monitoring_enable(PWR_MON_VCORE);
-		res = TEE_SUCCESS;
+	case INT_TAMP7:
+		if (IS_ENABLED(CFG_STM32MP21)) {
+			/* ADC2 (adc2_awd1) analog watchdog monitoring 1 */
+			res = TEE_SUCCESS;
+			break;
+		} else if (IS_ENABLED(CFG_STM32MP23) ||
+			   IS_ENABLED(CFG_STM32MP25)) {
+			/* VDDCORE monitoring under/over voltage */
+			stm32mp_pwr_monitoring_enable(PWR_MON_VCORE);
+			res = TEE_SUCCESS;
+			break;
+		}
 		break;
 
-	case INT_TAMP12: /* VDDCPU (Cortex A35) monitoring under/over voltage */
-		stm32mp_pwr_monitoring_enable(PWR_MON_VCPU);
-		res = TEE_SUCCESS;
+	case INT_TAMP12:
+		if (IS_ENABLED(CFG_STM32MP21)) {
+			/* ADC2 (adc2_awd2) analog watchdog monitoring 2 */
+			res = TEE_SUCCESS;
+			break;
+		} else if (IS_ENABLED(CFG_STM32MP23) ||
+			   IS_ENABLED(CFG_STM32MP25)) {
+			/* VDDCPU (Cortex A35) monitoring under/over voltage */
+			stm32mp_pwr_monitoring_enable(PWR_MON_VCPU);
+			res = TEE_SUCCESS;
+			break;
+		}
+		break;
+
+	case INT_TAMP13:
+	case INT_TAMP16:
+		if (IS_ENABLED(CFG_STM32MP21))
+			res = TEE_SUCCESS;
 		break;
 
 	case INT_TAMP5:
@@ -346,7 +370,6 @@ TEE_Result stm32_activate_internal_tamper(int id)
 	case INT_TAMP9:
 	case INT_TAMP10:
 	case INT_TAMP11:
-	case INT_TAMP13:
 	case INT_TAMP14:
 	case INT_TAMP15:
 		res = TEE_SUCCESS;
-- 
2.47.2

